### PR TITLE
Add support for subset queries

### PR DESCRIPTION
**Pull request by jordoh**
This adds support for subset queries, i.e.:

```
query.filter(.subset("_id", .in, ids))
```
and
```
query.filter(.subset("_id", .notIn, ids))
```

### CODE REVIEW

**Review by Joannis — CHANGES_REQUESTED** [review]
I'd love a unit test to be added to cover this. Calling fluent with a subset query which then results in the expected MongoDB behaviour. Looks good otherwise

**Comment by jordoh** [comment]
Hi @Joannis - I've pushed with tests for `.in` and `.notIn`, but have a couple questions:

- I added a `Tester+Subset.swift` to match the pattern of existing tests, but it looks like that might belong in Fluent itself. Is it reasonable to include that here?

- I'm seeing the existing `testSoftDelete` fail locally on master (at ef1f069), and on CircleCI for this branch - it that expected? It doesn't look like that failed on CircleCI for ef1f069 in this repo.

**Comment by Joannis** [comment]
Hello @jordoh :)

First of all, thanks for the PR 👍 
The unit tests could be in Fluent, I think that wouldn't be unreasonable. @tanner0101 , what do you think?

I just noticed that, too. It's not your code that broke it, I'm certain of that.

I'll leave the merge and tag up to @tanner0101 , and if he thinks this test needs to be moved to Fluent that could be done after the merge :)

Thanks again :)

**Review by Joannis — APPROVED** [review]

**Pull request merged**
